foundations/getting/getting.tex

```diff
@@ -0,0 +1,498 @@
+\chapter{Getting OpenSSL}
+\label{chp:getting-openssl}
+
+There are multiple ways to get hold of OpenSSL. This chapter will cover
+how to compile and install OpenSSL from source, as well as discuss various
+options for obtaining OpenSSL in pre-built binary form.
+
+\section{Understanding OpenSSL version numbers}
+
+The OpenSSL project (\url{https://www.openssl.org}) develops and
+maintains the OpenSSL source code. The project stores all of the source
+code in a source code control system known as git. Periodically the
+project will generate a new \emph{release}. This is a snapshot of the source
+code at a given point in time that is collected together into a single archive
+file and made available for download.
+
+At any one time there may be multiple versions of OpenSSL that are receiving
+support by the project team. Support in this context means that bugs are
+fixed and security issues are investigated, reported and fixed as appropriate.
+
+For example OpenSSL 1.1.0 and OpenSSL 1.0.2 are two versions of OpenSSL. Each
+version of OpenSSL may have multiple releases associated with it. These are
+identified by a letter after the version number. The very first release for
+a version has no letter. For example the first release for OpenSSL 1.1.0 has
+the version number 1.1.0, the second one is 1.1.0a, the third one is 1.1.0b,
+and so on. The letter releases contain bug and security fixes for the given
+OpenSSL version (but never new features). To ensure that you always have the
+latest fixes you should always try to use the latest letter release that is
+available for the OpenSSL version that you are using. Letter releases for a
+version are always fully backward compatible.
```

Comment:
Oooooh!  The distinction between version and release, I never caught on to that!  Handy!